src/peergos/server/corenode/JdbcIpnsAndSocial.java

```diff
@@ -112,35 +142,38 @@ public boolean delete() {
 
     private volatile boolean isClosed;
 
-    public JdbcIpnsAndSocial(Connection conn) throws SQLException {
+    public JdbcIpnsAndSocial(Connection conn, SqlSupplier commands) throws SQLException {
         this.conn = conn;
-        init();
+        init(commands);
     }
 
-    private synchronized void init() throws SQLException {
+    private synchronized void init(SqlSupplier commands) throws SQLException {
         if (isClosed)
             return;
 
         //do tables exists?
-        Statement stmt = conn.createStatement();
-        ResultSet rs = stmt.executeQuery(TABLE_NAMES_SELECT_STMT);
+        PreparedStatement list = conn.prepareStatement(commands.listTablesCommand());
+        ResultSet rs = list.executeQuery();
 
-        ArrayList<String> missingTables = new ArrayList<>(TABLES.keySet());
+        List<String> tables = new ArrayList<>();
         while (rs.next()) {
-            String tableName = rs.getString("name");
-            missingTables.remove(tableName);
+            String tableName = rs.getString(1);
+            tables.add(tableName);
         }
 
-        for (String missingTable: missingTables) {
-            try {
+        try {
+            if (! tables.contains("followrequests")) {
```

Comment:
A few suggestions:
* arguably the DDL should live in a separate script (for both implementations), particularly since there is more to do now with the postgres impl. around user  setup that cannot be done in this client. 
*  if we do keep these DDL statement here we  can remove this conditional since both sqlite and  postgres support ``create table if not exists`` statements.

Reply:
create table if not exists would be great! I didn't know that existed